code/datums/elements/wheel.dm

```diff
@@ -0,0 +1,30 @@
+/// Element which spins you as you move
+/datum/element/wheel
+
+/datum/element/wheel/Attach(datum/target)
+	. = ..()
+	if(!ismovable(target))
+		return ELEMENT_INCOMPATIBLE
+	RegisterSignal(target, COMSIG_MOVABLE_MOVED, isliving(target) ? PROC_REF(on_living_moved) : PROC_REF(on_moved))
+
+/datum/element/wheel/Detach(datum/source)
+	. = ..()
+	UnregisterSignal(source, COMSIG_MOVABLE_MOVED)
+
+/datum/element/wheel/proc/on_living_moved(mob/living/target, atom/oldloc, direction, forced)
+	SIGNAL_HANDLER
+	if(forced || target.incapacitated() || target.body_position == LYING_DOWN || CHECK_MOVE_LOOP_FLAGS(target, MOVEMENT_LOOP_OUTSIDE_CONTROL))
+		return
+	on_moved(target, oldloc, direction, forced)
```

Comment:
Why have a second proc instead of just `if(isliving(target))` in the main proc?

Reply:
Good question